ui/src/views/tool/McpToolFormDrawer.vue

```
@@ -246,7 +246,10 @@ const submit = async (formEl: FormInstance | undefined) => {
   await formEl.validate((valid: any) => {
     if (valid) {
       try {
-        JSON.parse(form.value.code as string)
+        const parsed = JSON.parse(form.value.code as string)
+        if (typeof parsed !== 'object' || parsed === null || Array.isArray(parsed)) {
+          throw new Error('Code must be a valid JSON object')
+        }
       } catch (e) {
         MsgError(t('views.applicationWorkflow.nodes.mcpNode.mcpServerTip'))
         return
```

Here’s a review of your code:

The changes made are mostly improvements to handle invalid JSON input more effectively. Your original attempt to use `JSON.parse` might still fail silently if the user inputs malformed data. By catching this exception within an `if (!parsed)` block, you ensure that any error is explicitly logged with MessageError.

**Suggested Improvement:** Add a log message for debugging purposes if the JSON parsing fails. This will help in identifying why the content doesn't validate as expected during testing or maintenance.

```javascript
try {
  const parsed = JSON.parse(form.value.code as string);
  if (typeof parsed !== 'object' || parsed === null || Array.isArray(parsed)) {
    // Log the error here so it's easier to track when something goes wrong
    console.error('Invalid JSON structure:', e.message);
    throw new Error('Code must be a valid JSON object');
  }
} catch (e) {
  MsgError(t('views.applicationWorkflow.nodes.mcpNode.mcpServerTip'));
}
```

This approach provides clarity on what went wrong during JSON parsing and prevents silent failures.